lib/rules/require-curly-braces.js

```diff
@@ -97,14 +91,12 @@ module.exports.prototype = {
 
         if (typeIndex['if'] || typeIndex['else']) {
             file.iterateNodesByType('IfStatement', function(node) {
-                if (typeIndex.if && node.consequent.type !== 'BlockStatement') {
+                if (typeIndex['if'] && isNotABlockStatement(node.consequent)) {
                     addError('If', node);
                 }
-                if (typeIndex['else'] && node.alternate &&
-                    node.alternate.type !== 'BlockStatement' &&
-                    node.alternate.type !== 'IfStatement'
-                ) {
-                    addError('Else', node);
+                if (typeIndex['else'] && isNotABlockStatement(node.alternate) &&
+                    node.alternate.type !== 'IfStatement') {
```

Comment:
But that also means we should add a test for `else if` to this PR in the require-curly-braces suite so we don't lose coverage should we ever remove the preset tests.